### PR TITLE
Refactor: generalize `force/stress_symmetry` to `symmetrize_vec3/mat3`

### PR DESCRIPTION
**Pull request by maki49**
generalize previous `force_symmetry` and `stress_symmetry` to `symmetrize_vec3_nat` and `symmetrize_mat3`, for future use in `Relax` to implement #2901 (which needs further refactor of `Symmetry` and `UnitCell` so I will do that in another PR)

### CODE REVIEW

File: source/module_hamilt_pw/hamilt_stodft/sto_forces.cpp [hunk 1]
```diff
@@ -106,29 +106,8 @@ void Sto_Forces::cal_stoforce(ModuleBase::matrix& force,
...
-				{
-                    p_symm->check_translation(pos[iat * 3 + k], -floor(pos[iat * 3 + k]));
-                    p_symm->check_boundary(pos[iat * 3 + k]);
-                }
-				iat++;				
-			}
-		}
-		
+    {
+        double d1, d2, d3;
```
Comment (dyzheng) [inline source/module_hamilt_pw/hamilt_stodft/sto_forces.cpp:110]: There are some code deleted without annotation, @Qianruipku can you review this code block?

**Review by dyzheng — APPROVED** [review]

File: source/module_cell/module_symmetry/symmetry.h [hunk 2]
```diff
@@ -102,8 +102,8 @@ class Symmetry : public Symmetry_Basic
 	void rho_symmetry(double *rho, const int &nr1, const int &nr2, const int &nr3);
 	void rhog_symmetry(std::complex<double> *rhogtot, int* ixyz2ipw, const int &nx, 
 			const int &ny, const int &nz, const int & fftnx, const int &fftny, const int &fftnz);
-	void force_symmetry(ModuleBase::matrix &force, double* pos, const UnitCell &ucell);
-	void stress_symmetry(ModuleBase::matrix &sigma, const UnitCell &ucell);
+    void symmetrize_vec3_nat(double* v)const;
```
Comment (dyzheng) [inline source/module_cell/module_symmetry/symmetry.h:107]: Annotations should be added for functions in *.h files.

**Review by dyzheng — APPROVED** [review]

**Comment by dyzheng** [comment]
I think this PR is good, but at least a Issue or a existed project should be linked to this PR.

**Comment by maki49** [comment]
> I think this PR is good, but at least a Issue or a existed project should be linked to this PR.

#2901, but not completed. 
To do the next step we need to refactor `UnitCell` (#3024)

**Review by dyzheng — APPROVED** [review]

**Pull request merged**